### PR TITLE
Migrate HttpClient from 4.x to 5.x

### PR DESCRIPTION
**Pull request by fbruton**
`< before change`
`> after change`
```
46,48c46,48
< [INFO] +- org.apache.httpcomponents:httpclient:jar:4.5.13:compile
< [INFO] |  +- org.apache.httpcomponents:httpcore:jar:4.4.15:compile
< [INFO] |  \- commons-logging:commons-logging:jar:1.2:compile
---
> [INFO] +- org.apache.httpcomponents.client5:httpclient5:jar:5.2.1:compile
> [INFO] |  +- org.apache.httpcomponents.core5:httpcore5:jar:5.2.1:compile
> [INFO] |  \- org.apache.httpcomponents.core5:httpcore5-h2:jar:5.2:compile

```

### CODE REVIEW

**Review by drivenflywheel — APPROVED** [review]
LGTM 🚀

File: src/main/java/emissary/client/HTTPConnectionFactory.java [hunk 1]
```diff
@@ -96,7 +95,8 @@ private HTTPConnectionFactory() {
             final Configurator cfg = config == null ? ConfigUtil.getConfigInfo(HTTPConnectionFactory.class) : config;
             // if someone doesn't want keep alives...
             if (!cfg.findBooleanEntry(CFG_HTTP_KEEPALIVE, DFLT_KEEPALIVE)) {
-                this.connReuseStrategy = NoConnectionReuseStrategy.INSTANCE;
+                this.connReuseStrategy = DefaultConnectionReuseStrategy.INSTANCE;
```
Comment (ldhardy) [inline src/main/java/emissary/client/HTTPConnectionFactory.java:98]: This change is probably fine, but I'm just noting here.
We're moving from a "A strategy that never re-uses a connection" posture to a more subtle re-use policy, where sometimes connections will be reused.  Again, I think it's probably safe to make this change but if it introduces problems, they will probably be tricky to track down to this change.  The strategy we changed to does the following:

```
Default implementation of a strategy deciding about connection re-use. The strategy determines whether a connection is persistent or not based on the message’s protocol version and Connection header field if present. Connections will not be re-used and will close if any of these conditions is met

- the close connection option is present in the request message
- the response message content body is incorrectly or ambiguously delineated
- the close connection option is present in the response message
- If the received protocol is HTTP/1.0 (or earlier) and keep-alive connection option is not present

In the absence of a Connection header field, the non-standard but commonly used Proxy-Connection 
header field will be used instead. If no connection options are explicitly given the default policy for the 
HTTP version is applied.  HTTP/1.1 (or later) connections are re-used by default. HTTP/1.0 (or earlier)
 connections are not re-used by default.
```

File: src/main/java/emissary/server/mvc/adapters/WorkSpaceAdapter.java [hunk 2]
```diff
@@ -46,7 +47,8 @@ public EmissaryResponse outboundOpenWorkSpace(final String place, final String s
         nvps.add(new BasicNameValuePair(SPACE_NAME, space));
         method.setEntity(new UrlEncodedFormEntity(nvps, StandardCharsets.UTF_8));
         // set a timeout in case a node is unresponsive
-        method.setConfig(RequestConfig.custom().setConnectTimeout(60000).setSocketTimeout(60000).build());
+        method.setConfig(RequestConfig.custom().setConnectTimeout(Timeout.ofMilliseconds(60000)).setConnectionKeepAlive(Timeout.ofMilliseconds(60000))
```
Comment (ldhardy) [inline src/main/java/emissary/server/mvc/adapters/WorkSpaceAdapter.java:50]: Another deprecated custom.setConnectTimeout

File: src/test/java/emissary/client/EmissaryClientTest.java [hunk 3]
```diff
@@ -60,9 +61,9 @@ void testDefaultRequestConfigFromClasspath() {
         RequestConfig requestConfig = client.getRequestConfig();
         // asserted values are copied from src/main/resource/emissary/client/EmissaryClient.cfg, which is packaged
         // with the jar
-        assertEquals(TimeUnit.MINUTES.toMillis(10), requestConfig.getConnectTimeout());
-        assertEquals(TimeUnit.MINUTES.toMillis(5), requestConfig.getConnectionRequestTimeout());
-        assertEquals(TimeUnit.SECONDS.toMillis(90), requestConfig.getSocketTimeout());
+        assertEquals(Timeout.ofMilliseconds(TimeUnit.MINUTES.toMillis(10)), requestConfig.getConnectTimeout());
+        assertEquals(Timeout.ofMicroseconds(TimeUnit.MINUTES.toMillis(5)), requestConfig.getConnectionRequestTimeout());
```
Comment (ldhardy) [inline src/test/java/emissary/client/EmissaryClientTest.java:65]: A seeming inconsistency here between milli and micro?

File: src/main/java/emissary/client/EmissaryClient.java [hunk 4]
```diff
@@ -124,22 +126,22 @@ protected static void configure() {
...
         }
 
         staticRequestConfig =
-                RequestConfig.custom().setConnectTimeout(connectionTimeout)
-                        .setConnectionRequestTimeout(connectionManagerTimeout)
-                        .setSocketTimeout(socketTimeout)
-                        .setTargetPreferredAuthSchemes(Collections.singleton(AuthSchemes.DIGEST))
-                        .setProxyPreferredAuthSchemes(Collections.singleton(AuthSchemes.DIGEST))
+                RequestConfig.custom().setConnectTimeout(Timeout.ofMilliseconds(connectionTimeout))
+                        .setConnectionRequestTimeout(Timeout.ofMicroseconds(connectionManagerTimeout))
```
[the remark this answers is not in the document]
Comment (fbruton) [reply src/main/java/emissary/client/EmissaryClient.java:141]: The original code set microseconds, but there is a comment that states milliseconds. That leads me to believe that setting to microseconds was by accident. I can update to milliseconds to be consistent.

**Review by ldhardy — APPROVED** [review]

**Review by jpdahlke — APPROVED** [review]

**Comment by jpdahlke** [comment]
- [ ] confirm availability of artifacts

**Pull request merged**